erpnext/stock/doctype/delivery_trip/delivery_trip.js

```diff
@@ -66,6 +66,49 @@ frappe.ui.form.on('Delivery Trip', {
 		}
 	},
 
+	validate: function (frm) {
+		if (frm.doc.delivery_stops) {
+			frappe.call({
+				method: "erpnext.stock.doctype.delivery_trip.delivery_trip.validate_unique_delivery_notes",
+				args: {
+					delivery_stops: frm.doc.delivery_stops
+				},
+				async: false,
```

Comment:
Question...

Why `{ async: false }` ?

This will freeze the browser waiting for data to return, just makes things sluggish(even if just waiting for stuff)

Reply:
Oh, that was from the older commit.

The problem was that the `frappe.confirm` showed the prompt nicely, but the `frappe.call` continued, and validated the document anyway - so the prompt would stay hanging, and the document was saved.

I was just testing the `freeze` and `async` params to see if I can somehow pause execution until user input.

Eventually had to do the hack to postprocess deletion, in case the user decides to not continue.

Reply:
haha... yeah just saw it was an old commit my bad

Reply:
Ah, no problem. Do tell me if you find a way to do that though. I'm still using a hack 😅